src/ruvsarpur.py

```diff
@@ -258,39 +258,52 @@ def parseArguments():
   parser.add_argument("--desc", help="Displays show description text when available", action="store_true")
   
   parser.add_argument("-d", "--debug", help="Prints out extra debugging information while script is running", action="store_true")
+
+  parser.add_argument("-p","--portable", help="Saves the tv schedule and the download log in the current directory instead of {0}".format(LOG_DIR), action="store_true")
   
   return parser.parse_args()
  
+# Appends the config directory to config file names
+def appendConfigDirectory(portable, file_name):
+  if portable :
+    return "./{0}".format(file_name)
+  else:
+    return "{0}/{1}".format(LOG_DIR,file_name)
+
+
 # Saves a list of program ids to a file
-def savePreviouslyRecordedShows(pids):
+def savePreviouslyRecordedShows(pids,portable):
   #make sure that the directory exists
-  os.makedirs(os.path.dirname(PREV_LOG_FILE), exist_ok=True)
+  rec_file = appendConfigDirectory(portable, PREV_LOG_FILE)
+  os.makedirs(os.path.dirname(rec_file), exist_ok=True)
 
-  with open(PREV_LOG_FILE, 'w+') as thefile:
+  with open(rec_file, 'w+') as theFile:
     for item in pids:
-      thefile.write("%s\n" % item)
+      theFile.write("%s\n" % item)
 
 # Gets a list of program ids from a file
-def getPreviouslyRecordedShows():
-  rec_file = Path(PREV_LOG_FILE)
+def getPreviouslyRecordedShows(portable):
+  rec_file = Path(appendConfigDirectory(portable, PREV_LOG_FILE))
   if rec_file.is_file():
     lines = [line.rstrip('\n') for line in rec_file.open('r+')]
     return lines
   else:
     return []
 
-def saveCurrentTvSchedule(schedule):
+def saveCurrentTvSchedule(schedule,portable):
   # Format the date field
   schedule['date'] = schedule['date'].strftime('%Y-%m-%d')
 
+  tv_file = appendConfigDirectory(portable,TV_SCHEDULE_LOG_FILE)
+
   #make sure that the log directory exists
-  os.makedirs(os.path.dirname(TV_SCHEDULE_LOG_FILE), exist_ok=True)
+  os.makedirs(os.path.dirname(tv_file), exist_ok=True)
```

Comment:
Same comment as in an earlier review, consider explicitly checking for the path existense instead of using exists_ok. There was an issue with this parameter in earlier python 3 versions.